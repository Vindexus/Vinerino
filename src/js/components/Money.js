import React from 'react'

export default class Money extends React.Component {
  render () {
    const totalCents = this.props.amount
    let display = 0

    if(!totalCents) {
      return <span class="money money-zero">$0</span>
    }

    const dollars = Math.floor(totalCents / 100)
    const cents = totalCents - (dollars * 100)

    if(dollars >= 1000) {
      let ks = parseFloat((dollars / 1000).toFixed(2))
      display = ks + 'k'
    }
    else if(dollars >= 10) {
      display = parseFloat((dollars+cents).toFixed(2))
    }
    else {
      display = dollars.toString() + '.' + (cents < 10 ? '0' + cents.toString() : cents).toString()
    }

    return <span class="money">${display}</span>
  }
}