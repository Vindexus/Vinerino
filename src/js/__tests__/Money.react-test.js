// Link.react-test.js
import React from 'react';
import Money from '../Components/Money';
import renderer from 'react-test-renderer';

import {shallow} from 'enzyme';

test('10 cents', () => {
  const component = renderer.create(
    <Money amount={10} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  let money = shallow(
    <Money amount={10} />
  );

  expect(money.text()).toEqual('$0.10');
})

test('100 cents', () => {
  const component = renderer.create(
    <Money amount={100} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  let money = shallow(
    <Money amount={100} />
  );

  expect(money.text()).toEqual('$1.00');
})


test('101 cents', () => {
  const component = renderer.create(
    <Money amount={101} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  let money = shallow(
    <Money amount={101} />
  );

  expect(money.text()).toEqual('$1.01');
})

test('110 cents', () => {
  const component = renderer.create(
    <Money amount={110} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  let money = shallow(
    <Money amount={110} />
  );

  expect(money.text()).toEqual('$1.10');
})

test('1000 cents', () => {
  const component = renderer.create(
    <Money amount={1000} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  let money = shallow(
    <Money amount={1000} />
  );

  expect(money.text()).toEqual('$10');
})

test('10000 cents', () => {
  const component = renderer.create(
    <Money amount={10000} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  let money = shallow(
    <Money amount={10000} />
  );

  expect(money.text()).toEqual('$100');
})


test('100000 cents', () => {
  const component = renderer.create(
    <Money amount={100000} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  let money = shallow(
    <Money amount={100000} />
  );

  expect(money.text()).toEqual('$1k');
})
