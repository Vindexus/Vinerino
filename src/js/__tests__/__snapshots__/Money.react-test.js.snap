// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`10 cents 1`] = `
<span
  className="money"
>
  $
  0.10
</span>
`;

exports[`100 cents 1`] = `
<span
  className="money"
>
  $
  1.00
</span>
`;

exports[`101 cents 1`] = `
<span
  className="money"
>
  $
  1.01
</span>
`;

exports[`110 cents 1`] = `
<span
  className="money"
>
  $
  1.10
</span>
`;

exports[`1000 cents 1`] = `
<span
  className="money"
>
  $
  10
</span>
`;

exports[`10000 cents 1`] = `
<span
  className="money"
>
  $
  100
</span>
`;

exports[`100000 cents 1`] = `
<span
  className="money"
>
  $
  1k
</span>
`;
